flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/enumerator/tracker/SplitTracker.java

```diff
@@ -159,12 +170,35 @@ public boolean isAssigned(String splitId) {
         return assignedSplits.contains(splitId);
     }
 
+    /**
+     * Function to get children splits available for given parent ids. This will ensure not to
+     * iterate all the values in knownSplits so saving compute
+     */
+    public List<DynamoDbStreamsShardSplit> splitsAvailableForAssignment(
+            Set<String> parentSplitIds) {
+        return parentSplitIds
+                .parallelStream()
+                .filter(parentChildSplitMap::containsKey)
+                .filter(
+                        splitId -> {
+                            boolean splitIsNotAssigned = !isAssigned(splitId);
+                            DynamoDbStreamsShardSplit split = knownSplits.get(splitId);
+                            return splitIsNotAssigned
+                                    && !isFinished(splitId)
+                                    && verifyParentIsEitherFinishedOrCleanedUp(split);
```

Comment:
can we extract this to a private method? Prevent code duplication for this essential logic.